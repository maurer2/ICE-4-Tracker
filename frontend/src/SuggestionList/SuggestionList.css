.suggestions {
	display: none;
	margin: 0;
	padding: 0;
	background: #fff;
	list-style: none;
  text-align: left;
  flex-grow: 1;
}

.suggestions--are-visible {
	display: block;
}

.suggestion {
  padding: 0 0.5rem;
}

.suggestion--is-active {
  background: red;
}

.button {
  border: 0;
  background: none;
  text-align: left;
}
