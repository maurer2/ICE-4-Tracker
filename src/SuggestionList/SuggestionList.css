.suggestions {
	display: none;
	margin: 0;
	padding: 0;
	background: #fff;
	list-style: none;
  text-align: left;
  flex-grow: 1;
}

.suggestions--are-visible {
	display: block;
}

.suggestion {
  padding: 0.5rem;
}

.suggestion--is-active {
  background: #6b6b6b;
  color: #fff;
}

.button {
  display: block;
  border: 0;
  text-align: left;
  background: none;
  color: inherit;
}
