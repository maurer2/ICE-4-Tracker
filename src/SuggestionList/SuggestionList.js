import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import styles from './SuggestionList.module.css';

import Suggestion from '../Suggestion/Suggestion';

const SuggestionList = (props) => {
  const {
    showSuggestions, trainNumbers, selectedTrainNumber, cbHandleClickEvent,
  } = props;

  return (
    <ul
      className={ classNames(
        styles.suggestions,
        styles.search, {
          [styles['suggestions--are-visible']]: showSuggestions,
        },
      )}
    >
      {trainNumbers.map((trainNumber, index) => (
        <Suggestion
          isActive={selectedTrainNumber === trainNumber}
          trainNumber={trainNumber}
          cbHandleClickEvent={cbHandleClickEvent}
          key={index}
        />
      ))}
    </ul>
  );
};

export default SuggestionList;

SuggestionList.propTypes = {
  showSuggestions: PropTypes.bool.isRequired,
  trainNumbers: PropTypes.array.isRequired,
  selectedTrainNumber: PropTypes.string.isRequired,
  cbHandleClickEvent: PropTypes.func.isRequired,
};
