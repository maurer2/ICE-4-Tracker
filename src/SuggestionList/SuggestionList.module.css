.suggestions {
	display: none;
	margin: 0;
	padding: 0;
	background: #fff;
	list-style: none;
  text-align: left;
  flex-grow: 1;
}

.suggestions--are-visible {
	display: block;
}
