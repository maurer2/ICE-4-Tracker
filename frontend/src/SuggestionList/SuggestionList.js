import React, { Component } from 'react';
import './SuggestionList.css';

class SuggestionList extends Component {
  render() {
    return (
      <ul className={ `suggestions ${ (this.props.showSuggestions) ? 'suggestions--are-visible' : '' }` }>
        { this.props.trainNumbers.map((trainNumber, index) =>
          <li className={`suggestion ${ this.props.selectedTrainNumber === trainNumber
            ? 'suggestion--is-active' : '' }`} key={ index }>
              { trainNumber }
          </li>)
        }
      </ul>
    )
  }
}

export default SuggestionList;
