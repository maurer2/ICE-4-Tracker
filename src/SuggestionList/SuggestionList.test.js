import React from 'react';
import SuggestionList from './SuggestionList';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

describe('SuggestionList', () => {
  const mockedHandleClick = jest.fn();

  const wrapper = shallow(
    <SuggestionList
      showSuggestions={true}
      trainNumbers={['500', '600', '700']}
      selectedTrainNumber={'500'}
      cbHandleClickEvent={mockedHandleClick}
    />
  );

  it('SuggestionList should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
