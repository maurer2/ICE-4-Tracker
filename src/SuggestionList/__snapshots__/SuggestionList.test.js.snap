// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuggestionList SuggestionList should match snapshot 1`] = `
<ul
  className="suggestions search suggestions--are-visible"
>
  <Suggestion
    cbHandleClickEvent={[MockFunction]}
    isActive={true}
    key="0"
    trainNumber="500"
  />
  <Suggestion
    cbHandleClickEvent={[MockFunction]}
    isActive={false}
    key="1"
    trainNumber="600"
  />
  <Suggestion
    cbHandleClickEvent={[MockFunction]}
    isActive={false}
    key="2"
    trainNumber="700"
  />
</ul>
`;
